client/src/modules/list/components/Items/InputBar/InputBar.jsx

```diff
@@ -67,57 +65,43 @@ class InputBar extends Component {
       {
         itemName: value
       },
-      () => {
-        this.handleAddButtonStatus();
-        this.handleTipVisibility();
-      }
+      () => this.validateName(this.handleAddButtonStatus)
     );
   };
 
   handleAddButtonStatus = () => {
-    const { itemName } = this.state;
+    const { errorMessageId, itemName } = this.state;
 
-    this.setState({ isButtonDisabled: !itemName });
+    this.setState({ isButtonDisabled: !itemName || errorMessageId });
   };
 
-  handleTipVisibility = () => {
-    const { itemName } = this.state;
-    const isItemNameEmpty = !_trim(itemName);
-
-    if (isItemNameEmpty) {
-      this.setState({ isTipVisible: true });
-    }
-  };
-
-  hideTipAfterTimeout = () =>
-    setTimeout(() => this.setState({ isTipVisible: false }), 5000);
-
-  handleFormSubmit = event => {
-    event.preventDefault();
-
+  handleAddItem = () => {
     const {
       addItem,
       currentUser,
       match: {
         params: { id }
       }
     } = this.props;
-    const { itemName } = this.state;
+    const { errorMessageId, itemName } = this.state;
     const newItem = {
       authorId: currentUser.id,
       name: itemName
     };
 
-    if (_trim(itemName)) {
+    if (!errorMessageId) {
       this.setState({ pending: true });
 
       return addItem(newItem, id).finally(() => {
```

Comment:
consider refactoring this to try catch block

Reply:
Changing to `try-catch`  block is a good idea but I think we should avoid any refactoring unless it is necessary for the given situation